test/DGmethods/cfl.jl

```diff
@@ -0,0 +1,149 @@
+using Test
+using MPI
+using CLIMA
+using CLIMA.Mesh.Topologies
+using CLIMA.Mesh.Grids
+using CLIMA.Mesh.Grids: VerticalDirection, HorizontalDirection, EveryDirection
+using CLIMA.VTK
+using Logging
+using Printf
+using LinearAlgebra
+using CLIMA.DGmethods: DGModel, init_ode_state, LocalGeometry, cfl
+using CLIMA.DGmethods.NumericalFluxes: Rusanov, CentralGradPenalty,
+                                       CentralNumericalFluxDiffusive
+
+using CLIMA.PlanetParameters: kappa_d
+using CLIMA.Atmos: AtmosModel,
+                   AtmosAcousticLinearModel, RemainderModel,
+                   NoOrientation,
+                   NoReferenceState, ReferenceState,
+                   DryModel, NoRadiation, NoSubsidence, PeriodicBC,
+                   Gravity, HydrostaticState, IsothermalProfile,
+                   ConstantViscosityWithDivergence, vars_state, soundspeed
+
+const ArrayType = CLIMA.array_type()
+
+using CLIMA.MoistThermodynamics: air_density, total_energy, internal_energy,
+                                 soundspeed_air
+
+using CLIMA.VariableTemplates: Vars
+using StaticArrays
+
+function initialcondition!(state, aux, coords, t)
+  FT = eltype(state)
+
+  p∞::FT = 10 ^ 5
+  T∞::FT = 300
+  translation_speed::FT = 150
+  translation_angle::FT = pi / 4
+  α = translation_angle
+  u∞ = SVector(translation_speed * cos(α), translation_speed * sin(α), 0)
+
+  u = u∞
+  T = T∞
+  # adiabatic/isentropic relation
+  p = p∞ * (T / T∞) ^ (FT(1) / kappa_d)
```

Comment:
@tapios I know that this is just a test, but maybe we can add `struct IsentropicProcess end` and relevant functions to MoistThermo (`pressure(p∞,T,T∞,::IsentropicProcess) p∞ * (T / T∞) ^ (FT(1) / kappa_d)`).

Reply:
Sure. That would be good. We'll need them also for some aspects of the boundary conditions (see design doc), then with the proper moist generalizations where `1/kappa_d` becomes `c_pm/R_m`.

Reply:
You'd want to be a bit careful with `isentropic`, though. These relations really only hold for *unsaturated* isentropic processes.